src/zcash/Address.hpp

```diff
@@ -97,6 +99,116 @@ typedef boost::variant<InvalidEncoding, SproutPaymentAddress> PaymentAddress;
 typedef boost::variant<InvalidEncoding, SproutViewingKey> ViewingKey;
 typedef boost::variant<InvalidEncoding, SproutSpendingKey> SpendingKey;
 
+//! Sapling functions. 
+class SaplingPaymentAddress {
+public:
+    diversifier_t d;
+    uint256 pk_d;
+
+    SaplingPaymentAddress() : d(), pk_d() { }
+    SaplingPaymentAddress(diversifier_t d, uint256 pk_d) : d(d), pk_d(pk_d) { }
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(d);
+        READWRITE(pk_d);
```

Comment:
@daira did we decide whether we would do something else with the serialization of SaplingPaymentAddress, or are we leaving the diversifer at the start?

Reply:
The spec is frozen.